### ISSUE TITLE
qm/dispatch/... and qm/process hooks

### ISSUE THREAD

**Issue by rask** [johnbillion/query-monitor#177]
As `qm/process` is seemingly deprecated (contains a backcompat notice above the hook), I though some more documentation was needed on the `qm/dispatch/<type>` hooks. One use of the `qm/process` hook was to disable output on some pages (e.g. if the admin bar was not available). How do I use the `qm/dispatch/...` hooks to disable query monitor output?

I'll dig through the code but a short documentation on the hooks would be nice. :)

**Comment by johnbillion** [comment]
Yeah documentation for the hooks in QM is on my todo list. You can use the `qm/dispatch/html` filter to disable all HTML output on a given page:

```
add_filter( 'qm/dispatch/html', function( $dispatch ) {
    if ( $condition ) {
        return false;
    } else {
        return $dispatch;
    }
} );
```

Similarly, there are `qm/dispatch/ajax`, `qm/dispatch/redirect`, and `qm/dispatch/rest` filters to disable the header output in AJAX requests, redirects, and REST API requests respectively, if necessary.

**Comment by rask** [comment]
`qm/dispatch/html` did the trick for me, I'll need to checkout how the AJAX etc. dispatcher are wired to find possible use cases.
